Remaining Tensor API fixes(CPU) - dims() -> sizes()

jerryzh168 opened the pull request:
Summary: Remaining fixes so that D12812029 will compile

Differential Revision: D13535559

ezyang commented:
@jerryzh168 This diff seems to do more than just rename dims() to sizes()?

ezyang commented on caffe2/mobile/contrib/ios/ios_caffe_predictor.cc:
```diff
@@ -50,8 +50,7 @@ Caffe2IOSPredictor::Caffe2IOSPredictor(const caffe2::NetDef& init_net,
   FLAGS_caffe2_force_shared_col_buffer = true;
-  caffe2::Tensor input(caffe2::CPU);
-  input.Resize(inData.dims);
+  caffe2::Tensor input(inData.dims, caffe2::CPU);
```
Is there any particular reason we can't rewrite these in the factory function style, e.g.,

```
auto input = empty(inData.dims);
```

jerryzh168 replied:
will change to caffe2::empty

ezyang commented on caffe2/mobile/contrib/ios/mpscnn/mpscnn.mm:
```diff
@@ -256,12 +256,12 @@ void computeOutputHW(
     int* OW) {
-  Tensor input(CPU), output(CPU);
-  input.Resize(1, 1, H, W);
+  std::vector<int64_t> dims = {1, 1, H, W};
```
I don't know if the constructor can handle initializer lists, but I know that empty definitely can:

```
Tensor input = empty({1, 1, H, W});
```

No need to allocate an entire vector here.

jerryzh168 replied:
looks like Tensor here is just used for record sizes, no actual memory is allocated...

ezyang commented on caffe2/queue/rebatching_queue.cc:
```diff
@@ -42,7 +42,7 @@ void concat(
       CAFFE_ENFORCE_EQ(inputZero[j].ndim(), input.dim());
       for (int k = 0; k < input.dim(); ++k) {
-        CAFFE_ENFORCE_EQ(input.sizes()[k], inputZero[j].dims()[k]);
+        CAFFE_ENFORCE_EQ(input.sizes()[k], inputZero[j].size(k));
```
good

ezyang commented:
Everything else looks ok

jerryzh168 commented:
> @jerryzh168 This diff seems to do more than just rename dims() to sizes()?

Yeah, but it's mostly renaming dims() to sizes()...

ezyang commented on caffe2/mobile/contrib/ios/ios_caffe_predictor.cc:
```diff
@@ -50,8 +50,7 @@ Caffe2IOSPredictor::Caffe2IOSPredictor(const caffe2::NetDef& init_net,
   FLAGS_caffe2_force_shared_col_buffer = true;
-  caffe2::Tensor input(caffe2::CPU);
-  input.Resize(inData.dims);
+  caffe2::Tensor input = caffe2::empty(inData.dims, at::dtype<uint8_t>().device(caffe2::CPU));
```
hum, at some point we should standardize on using kCPU, probably (not for this PR)

jerryzh168 committed:
Remaining Tensor API fixes - dims() -> sizes() (#15743)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/15743

Remaining fixes so that D12812029 will compile

Reviewed By: dzhulgakov

Differential Revision: D13535559

fbshipit-source-id: 6c4dd86a0c71444a2cf6f455b305a66551f32f41
